pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/StoolUploadRoute.java

```diff
@@ -0,0 +1,193 @@
+package org.broadinstitute.dsm.route;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import javax.servlet.http.HttpServletRequest;
+
+import com.google.gson.Gson;
+import lombok.extern.log4j.Log4j2;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.exception.FileColumnMissing;
+import org.broadinstitute.dsm.exception.FileWrongSeparator;
+import org.broadinstitute.dsm.exception.UploadLineException;
+import org.broadinstitute.dsm.model.StoolUploadObject;
+import org.broadinstitute.dsm.security.RequestHandler;
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.dsm.statics.RoutePath;
+import org.broadinstitute.dsm.statics.UserErrorMessages;
+import org.broadinstitute.dsm.util.SystemUtil;
+import org.broadinstitute.dsm.util.UserUtil;
+import org.broadinstitute.lddp.handlers.util.Result;
+import spark.QueryParamsMap;
+import spark.Request;
+import spark.Response;
+
+@Log4j2
+public class StoolUploadRoute extends RequestHandler {
+
+    private static final String PARTICIPANT_ID = "participantId";
+    private static final String MF_BARCODE = "mfBarcode";
+    private static final String RECEIVE_DATE = "receiveDate";
+
+    @Override
+    protected Object processRequest(Request request, Response response, String userId) throws Exception {
+        QueryParamsMap queryParams = request.queryMap();
+        String realm;
+
+        if (queryParams.value(RoutePath.REALM) != null) {
+            realm = queryParams.get(RoutePath.REALM).value();
+        } else {
+            throw new RuntimeException("No realm query param was sent");
+        }
+
+        String userIdRequest = UserUtil.getUserId(request);
+
+        if (UserUtil.checkUserAccess(realm, userId, "kit_upload", userIdRequest)) {
```

Comment:
I wonder if we should add a new role for that @pegahtah what is your feeling about that?